examples/playground/src/App.tsx

```diff
@@ -174,72 +208,42 @@ function App() {
     type: 'object',
   };
 
-  const [isEditor, setIsEditor] = useState(true);
-
-  const toggleEditorMode = () => setIsEditor(!isEditor);
+  const formData = {};
 
-  const translations = {
-    en: {
-      root_description: 'Desc registration form',
-      root_section_checkboxes_description: 'Checkboxes description',
-      root_section_checkboxes_items_0: 'item 0 en',
-      root_section_checkboxes_items_1: 'item 1 en',
-      root_section_checkboxes_items_2: 'item 2 en',
-      root_section_checkboxes_title: 'Checkboxes title',
-      root_section_select_description: 'Description select',
-      root_section_select_enum_0: 'enum 0 en',
-      root_section_select_enum_1: 'enum 1 en',
-      root_section_select_enum_2: 'enum 2 en',
-      root_section_select_title: 'Select title',
-      root_title: 'A registration form',
-    },
-    fr: {
-      root_description: "Description formulaire d'inscription",
-      root_section_checkboxes_description: 'Description checkboxes',
-      root_section_checkboxes_items_0: 'item 0 fr',
-      root_section_checkboxes_items_1: 'item 1 fr',
-      root_section_checkboxes_items_2: 'item 2 fr',
-      root_section_checkboxes_title: 'Titre checkboxes',
-      root_section_select_description: 'Description select',
-      root_section_select_enum_0: 'enum 0 fr',
-      root_section_select_enum_1: 'enum 1 fr',
-      root_section_select_enum_2: 'enum 2 fr',
-      root_section_select_title: 'Titre select',
-      root_title: 'Formulaire',
-    },
+  const widgets = {
+    checkboxWidget: CheckboxWidget,
+    numberWidget: NumberWidget,
+    passwordWidget: PasswordWidget,
+    quizWidget: QuizWidget,
+    selectWidget: SelectWidget,
+    textWidget: TextWidget,
+    textareaWidget: TextareaWidget,
   };
 
-  const translationUi = {
-    'components.EditorFieldTemplate.parameters': 'Paramètres',
-    'components.WidgetPropertiesForm.description': 'Description',
-    'components.WidgetPropertiesForm.enum': 'Options',
-    'components.WidgetPropertiesForm.items': 'Options avancées',
-    'components.WidgetPropertiesForm.maxLength': 'Longueur maximum',
-    'components.WidgetPropertiesForm.maximum': 'Maximum',
-    'components.WidgetPropertiesForm.minLength': 'Longueur minimum',
-    'components.WidgetPropertiesForm.minimum': 'Minimum',
-    'components.WidgetPropertiesForm.required': 'Requis',
-    'components.WidgetPropertiesForm.submitWidgetParameter': 'Sauvegarder',
-    'components.WidgetPropertiesForm.title': 'Nom',
-  };
+  const [isEditor, setIsEditor] = useState(true);
+
+  const toggleEditorMode = () => setIsEditor(!isEditor);
```

Comment:
You prefer the state variables right before the JSX?

Reply:
Hmm.. in fact I do not have convention to order variables within a component, I just respect the global order 'variables - useEffect - render).

About variables, until a variable is used to declare another variable, I do not put hooks before local variable for example.